servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/Http2Exception.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.netty;
+
+import io.servicetalk.transport.api.RetryableException;
+
+import java.io.IOException;
+
+/**
+ * An exception that indicates <a href="https://tools.ietf.org/html/rfc7540">HTTP/2</a> protocol error.
+ */
+public class Http2Exception extends IOException {
```

Comment:
does this type need to be public?

Reply:
I made it public with a private ctor to allow users try-catch it (or check instanceof) without allowing them to create instances of this exception type.

Reply:
do we expect users to react differently to `Http2Exception` vs `IOException`? Currently `Http2Exception` doesn't expose any additional methods so it isn't clear when we would expect folks to know about the type.

Reply:
sounds reasonable, let me hide it for now